src/servers/grpc_server_v2.cc

```diff
@@ -1315,17 +1361,94 @@ InferGRPCToInput(
                 .c_str());
       }
 
-      // FIXMEV2 handle non-raw content types
+      // Try to read the raw contents if available
       const std::string& raw = io.contents().raw_contents();
       base = raw.c_str();
       byte_size = raw.size();
+
+      // Check the presence of explicit tensors
+      if (io.contents().bool_contents_size() != 0) {
+        RETURN_IF_ERR(InferGRPCToInputHelper(
+            io.name(), request.model_name(), "BOOL", io.datatype(), byte_size));
+        base = (const void*)io.contents().bool_contents().data();
+        byte_size = io.contents().bool_contents_size();
+      }
+
+      if (io.contents().int_contents_size() != 0) {
+        RETURN_IF_ERR(InferGRPCToInputHelper(
+            io.name(), request.model_name(), "INT32", io.datatype(),
```

Comment:
INT32 is used for int8, int16 and int32 and same for UINT32. protobuf doesn't have types of 16-bit and 8-bit integers. So for those types you will also need to reserialize like you do for strings.

Reply:
Done